qiskit/dagcircuit/_dagcircuit.py

```diff
@@ -554,7 +554,20 @@ def _map_condition(self, wire_map, condition):
             new_condition = (wire_map.get(bit0, bit0)[0], condition[1])
         return new_condition
 
-    def compose_back(self, input_circuit, wire_map=None):
+    def extends_at_the_end(self, dag, edge_map=None):
```

Comment:
this is highly similar to `compose_back()`. can we make the names similar too to indicate that?
```suggestion
    def extend_back(self, dag, edge_map=None):
```